src/shared/ui/form/tests/InputPassword.test.tsx

```diff
@@ -0,0 +1,194 @@
+import useAppForm from '@app/shared/lib/hooks/useAppForm';
+
+import InputPassword from '../InputPassword';
+
+/* Test helper functions */
+
+const getComponentByTestId = (
+  instance: renderer.ReactTestInstance,
+  testID: string,
+) => {
+  return instance.find(c => c.props.testID === testID).props;
+};
+
+const getPasswordRequirement = (
+  instance: renderer.ReactTestInstance,
+  requirementID: string,
+) => {
+  const requirement = getComponentByTestId(
+    instance,
+    `password_requirements:${requirementID}`,
+  );
+  return {
+    icon: requirement.children[0].props.children,
+    text: requirement.children[1].props.children,
+  };
+};
+
+const getPasswordRequirements = (instance: renderer.ReactTestInstance) => {
+  const minCharacters = getPasswordRequirement(instance, 'at_least_characters');
+
+  const capitalLetter = getPasswordRequirement(
+    instance,
+    'at_least_capital_letter',
+  );
+
+  const specialCharacters = getPasswordRequirement(
+    instance,
+    'no_special_characters',
+  );
+  const blankSpaces = getPasswordRequirement(instance, 'no_blank_spaces');
+
+  return { minCharacters, capitalLetter, specialCharacters, blankSpaces };
+};
+
+const FormTestWrapper = ({
+  children,
+  defaultValues = {},
+}: {
+  children: React.ReactNode;
+  defaultValues: any;
+}) => {
+  const { form } = useAppForm(SignUpFormSchema, {
+    defaultValues: {
+      email: '',
+      firstName: '',
+      lastName: '',
+      password: '',
+      ...defaultValues,
+    },
+    onSubmitSuccess: () => {},
+  });
+  return (
+    <TamaguiProvider>
+      <FormProvider {...form}>{children}</FormProvider>
+    </TamaguiProvider>
+  );
+};
+
+const createTest = ({
+  defaultValues = {},
+  displayRequirements = true,
+}: any = {}) => {
+  const tree = renderer.create(
+    <FormTestWrapper defaultValues={defaultValues}>
+      <InputPassword name="password" placeholder="Password" />
+    </FormTestWrapper>,
+  );
+  const instance = tree.root;
+  if (displayRequirements) {
+    const mEvent = { target: { value: '' } };
+    act(() => {
+      const input: any = instance.findByType(TextInput);
+      input.props.onFocus(mEvent);
+    });
+  }
+  return { tree, instance };
+};
+
+/* Test use-cases */
+
+describe('InputPassword', () => {
+  it('should render properly without displaying the password requirements', () => {
+    const { tree } = createTest({ displayRequirements: false });
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should render properly displaying the password requirements', () => {
+    const { tree } = createTest({ displayRequirements: true });
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should display validation when the password have 8 characters', () => {
+    expect(
+      getPasswordRequirements(
+        createTest({
+          defaultValues: {
+            password: 'testpassword',
+          },
+        }).instance,
+      ).minCharacters.icon.props.color,
+    ).toBe(colors.white);
```

Comment:
In general, relying on colour comparison to evaluate component behaviour doesn't quite feel ideal to me, for a couple reasons:

- Designs are often evolving, and different colours might be used to describe validity or non-validity states, making these tests possibly fragile
- In general, we should be keeping a11y top of mind whenever possible while writing unit tests, which to me often boils down to "how would a screen reader describe the screen state?" Screen readers generally don't care about the colour of text, but they do pay attention to a11y labels and other a11y props.

By thinking about how you'd navigate and get information about the screen state via, say, TalkBack (on Android) or VoiceOver (on iOS), it encourages us to write UIs that are accessible.

In this case, in the `PasswordRequirement` component, I might give it an a11y prop to indicate what state it is in. Maybe if the requirement is not met, I'd assign it `accessibilityState={{ disabled: true }}`, and then compare against that in my unit test. You could also use an `accessibilityLabel` to explain validation state in more words, but then you may need to add translation strings.

Having said all this, I also know that a11y can be tricky to get right and could take you down some deep rabbit holes, especially as Android and iOS don't perfectly align in behaviour. So I'm totally fine if you want to keep things the way you have them for now! It's still early on the project and a11y might not be a priority for a while yet.